### DISCUSSION TITLE
Welcome to sMQTTBroker Discussions!

### DISCUSSION THREAD

**Discussion by terrorsl** [terrorsl/sMQTTBroker#34] [Q&A] [answered]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JFRosado** [comment]
Hi:

   The examples are very few and simple, so i have a question. I want to run sMQTT in a esp32 to receive a topic. I then want to use the same esp32 to subscribe that topic and print the receive values. How can i do that?

Regards

**Reply by terrorsl** [reply]
Hi. You can only subscribe from an external client. The broker does not know how to subscribe to itself.

**Comment by ssjophus** [comment]
First of all Thank you terrorsl for giving us newbies a place to share and ask.
Is there a trick to getting this to work on an esp32 WROOM DA board. I always start a project with the base code, in this case the, example "simplebroker" and add to suit my specific project. In this case I haven't been able to get past the basics. The code seems to upload fine with no errors, however, I get no serial monitor feedback. No serial monitor errors. Nor does my router show any added connected devices. I of course edited the ssid and password fields to reflect my router credentials, as well as ensure that I have the proper com port, board type, and baud rate. I am using version 0.1.8. Normally in this situation where something simply does nothing, its because I missed, or fat fingered something. So I resort to talking to people smarter than me to straighten me back out. Any help would be appreciated.
Thank you,

**Reply by ssjophus** [reply]
Update: I added a debug line to the "while" loop to see if the code even uploaded properly.
```
while (WiFi.status() != WL_CONNECTED) { // Wait for the Wi-Fi to connect
        Serial.println("Connecting...");
        delay(1000);
    }
```
I get an output of "Connecting..." in the serial monitor now. So I know its uploading properly.

**Reply by terrorsl** [reply]
https://github.com/terrorsl/sMQTTBroker/issues/33#issuecomment-1368032106 in this thread how enable Debug log

**Reply by ssjophus** [reply]
The link in thread #33 gives me a 403 error:
" 403
Forbidden
Access to this resource on the server is denied! The owner of this website has blocked your visits.

This page is generated by [QUIC.cloud](https://quic.cloud/) CDN on the NA-US-IAD-89 server."

I will have to spend some time on youtube university, learning how to use debug anyhow. 
There has been another development in that if "Wifi.begin" comes after "Serial.begin" in the setup, anything between "Wifi.begin" and the while loop gets ignored.
ex: 
```
void setup()
{   
    const char* ssid = "(Redacted)";         // The SSID (name) of the Wi-Fi network you want to connect to
    const char* password = "(Redacted)"; // The password of the Wi-Fi network
   >Serial.begin(115200);
   >WiFi.begin(ssid, password);
    delay(10000);
    Serial.print("Connecting to ");
    Serial.print(ssid);

  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    Serial.print(".");
  }
```
The above code has no 10 sec delay and prints nothing to the serial monitor before it enters the while loop.

While:
```
void setup()
{   
    const char* ssid = "(Redacted)";         // The SSID (name) of the Wi-Fi network you want to connect to
    const char* password = "(Redacted)"; // The password of the Wi-Fi network
   >WiFi.begin(ssid, password);
   >Serial.begin(115200);
    delay(10000);
    Serial.print("Connecting to ");
    Serial.print(ssid);

  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    Serial.print(".");
  }
```
Gives me a 10 sec delay, then outputs The network it is connecting to, then enters the while loop, and loops forever.

I wonder if there is something board specific that's hanging up the wifi library. I did try putting "#include "Wifi.h" in the declarations thinking the wifi library might not be initializing properly but it made no difference.

**Reply by terrorsl** [reply]
https://www.mischianti.org/wp-content/uploads/2020/09/ArduinoIDE-esp32-log-levels.jpg

https://github.com/terrorsl/sMQTTBroker/issues/12#issuecomment-981882847

**Reply by terrorsl** [reply]
Also you can add WiFi.mode(WIFI_STA) before WiFi.begin, and in loop print status to know what's wrong

**Reply by ssjophus** [reply, marked as answer]
The new link is the same as the previous one, and I get the same error. However I added the wifi mode to station "WiFi.mode(WIFI_STA)" as per your advice and it seems to have fixed the issue.
Thank you!
